### PR TITLE
Implement feature - exclude table

### PR DESCRIPTION
**Pull request by bingluen**
Add query condition to exclude special table.

### CODE REVIEW

File: src/routes.php [hunk 1]
```diff
@@ -1,7 +1,13 @@
 <?php
 if (env('ENABLE_DOC_US', false)) {
     Route::get('schema', function (Illuminate\Http\Request $request) {
-        $schema = UniSharp\DocUs\Parser::getSchema();
+        $exclude = [];
+
+        if ($request->exclude !== null) {
+            $exclude = explode(',', strtolower($request->exclude));
```
Comment (youchenlee) [inline src/routes.php]: why with `strtolower`?

File: src/Parser.php [hunk 2]
```diff
@@ -7,17 +7,21 @@
...
+            ->map(function ($row) {
+                $table = head($row);
 
-            return [
-                'name' => $table,
-                'columns' => static::getColumns($table),
-                'indices' => static::getIndices($table),
-            ];
-        });
+                return [
```
Comment (youchenlee) [inline src/Parser.php]: unexpected changes

File: src/Parser.php [hunk 3]
```diff
@@ -7,17 +7,21 @@
 
 class Parser
 {
-    public static function getSchema()
+    public static function getSchema($exclude = [])
     {
-        return collect(DB::select('show tables'))->map(function ($row) {
-            $table = head($row);
+        return collect(DB::select('show tables'))
```
Comment (youchenlee) [inline src/Parser.php]: Please add one more `map` to prevent the redundant and implicit `head($row)`  statements.

File: tests/ParserTest.php [hunk 4]
```diff
@@ -146,4 +146,33 @@ public function testIndexKey()
...
+            return $table['name'];
+        })->toArray();
+
+        $this->assertEquals($expect, $tables);
+
+        DB::statement('DROP TABLE `tests`');
+
+        DB::statement('DROP TABLE `should_on_result`');
+    }
 }
```
Comment (youchenlee) [inline tests/ParserTest.php:204]: `exclude=table1,table2` is not covered by the test.

File: tests/ParserTest.php [hunk 5]
```diff
@@ -146,4 +146,33 @@ public function testIndexKey()
...
+            "CREATE TABLE `should_on_result` (
+                `tests` VARCHAR(1)
+            )"
+        );
+
+        $exclude = ['tests'];
+        $expect = ['should_on_result'];
+
+        $schema = Parser::getSchema($exclude);
+        $tables = $schema->map(function ($table) {
```
Comment (youchenlee) [inline tests/ParserTest.php]: perhaps `pluck()`

**Review by youchenlee — COMMENTED** [review]
I have left some comments inline. Thanks.

File: src/routes.php (hunk 1 above)
Comment (bingluen) [reply src/routes.php]: The query result of `show table` is lowercase.

I think that table should be exclude which table name same spelling but different case.

That why I call `strtolower`.

File: src/Parser.php (hunk 2 above)
Comment (bingluen) [reply src/Parser.php]: I try to change coding style to prevent this on my least commit a9e95d3

File: src/routes.php (hunk 1 above)
Comment (youchenlee) [reply src/routes.php]: might relate to MySQL `lower_case_table_names` config. 
the users should have the control.

Comment (bingluen) [reply src/routes.php]: Thanks for your reminder. I got it. I forgot to consider this configuration of MySQL.
I have removed `strtolower` on my last commit.

**Pull request merged**